examples/deepcopy-gen/generators/deepcopy.go

```diff
@@ -642,7 +681,7 @@ func (g *genDeepCopy) doMap(t *types.Type, sw *generator.SnippetWriter) {
 			sw.Do("for key := range *in {\n", nil)
 			sw.Do("(*out)[key] = struct{}{}\n", nil)
 			sw.Do("}\n", nil)
-		case t.Elem.IsAssignable():
+		case !hasPostDeepCopyMethod(t.Elem) && t.Elem.IsAssignable():
```

Comment:
Yes, I mix two orthogonal concept in this commit: postDeepCopy() and `skip-field` tag. Would make sense to split them apart in two commits indeed.

The reason for this line is: I want `postDeepCopy` to be private. If we use assignment and `val` has a `postDeepCopy` here, we have to call `postDeepCopy` from `t`'s `DeepCopyInto` func. If `t` and `val` are from different packages, this won't work. So we switch to  the mode where don't assign, but call `DeepCopy` on `val` (then the `postDeepCopy` is invoked from `val`'s `DeepCopyInto` func).

I will add a `PkgPath` comparison.